eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BitcoinCoreClient.scala

```diff
@@ -48,19 +48,24 @@ import scala.util.{Failure, Success, Try}
  * The Bitcoin Core client provides some high-level utility methods to interact with Bitcoin Core.
  *
  * @param rpcClient             bitcoin core JSON rpc client
- * @param onchainKeyManager_opt optional onchain key manager. If provided and its wallet name matches our rpcClient wallet's name, it will be used to sign transactions (it is assumed that bitcoin
- *                              core uses a watch-only wallet with descriptors generated by Eclair with this onchain key manager)
+ * @param onChainKeyManager_opt optional on-chain key manager. If provided it will be used to sign transactions (it is assumed that bitcoin
+ *                              core uses a watch-only wallet with descriptors generated by Eclair with this on-chain key manager)
  */
-class BitcoinCoreClient(val rpcClient: BitcoinJsonRPCClient, val onchainKeyManager_opt: Option[OnchainKeyManager] = None) extends OnChainWallet with Logging {
+class BitcoinCoreClient(val rpcClient: BitcoinJsonRPCClient, val onChainKeyManager_opt: Option[OnChainKeyManager] = None) extends OnChainWallet with Logging {
 
   import BitcoinCoreClient._
 
   implicit val formats: Formats = org.json4s.DefaultFormats
 
-  val useEclairSigner = onchainKeyManager_opt.exists(m => rpcClient.wallet.contains(m.wallet))
+  onChainKeyManager_opt.foreach { keyManager =>
+    require(rpcClient.wallet.contains(keyManager.wallet), s"eclair-backed bitcoin wallet mismatch: eclair-signer.conf uses wallet=${keyManager.wallet}, but eclair.conf uses wallet=${rpcClient.wallet.getOrElse("")}")
+  }
+
+  val useEclairSigner = onChainKeyManager_opt.nonEmpty
```

Comment:
Yes, that was explicitly mentioned in the commit message for https://github.com/ACINQ/eclair/pull/2726/commits/9263f40ad0a67e53a10a2801626f7ae517b04a8f

Supporting this case made the code very hard to follow, it wasn't tested and was quite bug-prone. I didn't know why that scenario in the first place, it really was hard to understand.

Can you provide more details about the exact flow you'd like to support, I'm still struggling to understand it? Then we'll see if we can support it in a simpler way than supporting this weird hybrid mode.

Reply:
If I'm understanding you correctly, can't this simply be handled in `Setup.scala`? In that scenario the user wants his eclair to keep using a `BitcoinCoreClient` with `onChainKeyManager_opt = None`, don't they? And since they provided an `eclair-signer.conf`, they should be able to get the descriptors (maybe needs small code changes to `Setup.scala`)?

Reply:
I went further and moved the creation of the eclair-backed wallet to the on-chain key manager in https://github.com/ACINQ/eclair/pull/2726/commits/221d257f8ae5e2e231d1c6ca0fe9eab9b7788fcc

The `createEclairBackedWallet` flow had always bothered me, we were jumping through a few hops of indirection and implicit wallet name matching which made it hard to reason about. I believe the intent is clearer now, and it should support the scenario you're mentioning?

Reply:
this is still not functionally equivalent to what we had before: if users add an `eclair-signer.conf` to their node's data directory, then on startup eclair will try to create an empty wallet and import descriptors even if its `bitcoind.wallet` sill points to a regular bitcoin wallet. If the timestamp in `eclair-signer.conf` is too old,. this will fail, eclair will not start and users will not be able to use eclair's api to generate descriptors and import them manually.

Reply:
Yes, that's true. But to be honest I think the issue is that it's really unclear to me what we are actually trying to support and why...if it was more clear what the spec actually is here, it would be easier to figure out what exactly to implement.

It looks like we're trying to support a lot of different use-cases that we're not going to use and users won't use either, which only makes the code more complex. Can we define the exact scope that we need instead and focus on only that?

Reply:
All the migration scenarios should now be supported with https://github.com/ACINQ/eclair/pull/2726/commits/eb49eba64bd69236bac8d1d1b8887a8b586e6598, let me know if you find any issue!